Makefile

```diff
@@ -1797,16 +1799,30 @@ $(ESP_COMBINED512): $(USER1_BIN) $(USER2_BIN)
 # IMPORTANT: this only works if DISABLE_LTO is defined, e.g. `DISABLE_LTO=1 make`
 topstrings: $(PARTIAL)
 	$(Q)for f in `find . -name \*.o`; do \
-	  str=$$($(OBJDUMP) -j .rodata.str1.4 -h $$f 2>/dev/null | \
-	    egrep -o 'rodata.str1.4 [0-9a-f]+' | \
+	  str=$$($(OBJDUMP) -j .rodata.str1.1 -j .rodata.str1.4 -h $$f 2>/dev/null | \
+	    egrep -o 'rodata.str1.. [0-9a-f]+' | \
 	    awk $$(expr match "$$(awk --version)" "GNU.*" >/dev/null && echo --non-decimal-data) \
 	      -e '{printf "%d\n", ("0x" $$2);}'); \
 	  [ "$$str" ] && echo "$$str $$f"; \
 	done | \
 	sort -rn >topstrings
 	$(Q)echo "Top 20 from ./topstrings:"
 	$(Q)head -20 topstrings
-	$(Q)echo "To get details: $(OBJDUMP) -j .rodata.str1.4 -s src/FILENAME.o"
+	$(Q)echo "To get details: $(OBJDUMP) -j .rodata.str1.1 -j .rodata.str1.4 -s src/FILENAME.o"
+
+# Same as topstrings but consider all read-only data
+topreadonly: $(PARTIAL)
```

Comment:
> Perhaps we could attempt an upstream patch?

I'm pretty sure they'd say 'No.'. Especially as it's ARM's code and you're suggesting adding random hacks to it in order to support another manufacturer's processors :)

> Is the config.h "out of the box"

No, that's been modified to include only what's needed. Feel free to play around and see, but I doubt you're going to get around the fact that the compiler needs telling to 

> Can this be done on a single declaration 

It might be tricker if it's static, as it will effectively be unnamed then :( There should still be some way though.

I believe you can use objcopy or something to put *everything from an entire object file into flash* - I'm sure I saw some ESP8266 hacks that did that. If the only stuff in those files is read-only then that could be an option.